python/ray/data/tests/test_stats.py

```diff
@@ -133,6 +135,76 @@ def test_dataset_stats_sort(ray_start_regular_shared):
     assert "sort_reduce" in stats, stats
 
 
+@patch("ray.data._internal.plan.logger")
+def test_dataset_stats_autolog(mock_logger, ray_start_regular_shared):
+    """Tests functionality to autolog stats after each stage execution,
+    enabled via the `DatasetContext.enable_auto_log_stats` parameter."""
+    context = DatasetContext.get_current()
+    context.enable_auto_log_stats = True
+    ds = ray.data.range(1000, parallelism=10)
+    # Check logged stats after each stage
+    ds = ds.map_batches(lambda x: x)
+    logger_args, logger_kwargs = mock_logger.info.call_args
+    assert (
+        canonicalize(logger_args[0])
+        == """Stage Z read: N/N blocks executed in T, N/N blocks split from parent
+* Remote wall time: T min, T max, T mean, T total
+* Remote cpu time: T min, T max, T mean, T total
+* Peak heap memory usage (MiB): N min, N max, N mean
+* Output num rows: N min, N max, N mean, N total
+* Output size bytes: N min, N max, N mean, N total
+* Tasks per node: N min, N max, N mean; N nodes used
+"""
+    )
+
+    ds = ds.map(lambda x: x)
+    logger_args, logger_kwargs = mock_logger.info.call_args
+    assert (
+        canonicalize(logger_args[0])
+        == """Stage N read->map_batches: N/N blocks executed in T
```

Comment:
shouldn't it be `map` instead of `read->map_batches`?

Reply:
same as above discussion https://github.com/ray-project/ray/pull/29876#discussion_r1012057494